controlplane/rosa/api/v1beta2/rosacontrolplane_types.go

```diff
@@ -48,8 +48,8 @@ type RosaControlPlaneSpec struct { //nolint: maligned
 	// The AWS Region the cluster lives in.
 	Region *string `json:"region"`
 
-	// Openshift version, for example "openshift-v4.14.5".
-	Version *string `json:"version"`
+	// Openshift version, for example "4.14.5".
+	Version string `json:"version"`
```

Comment:
We might want to add some validation here? We could check semver?

Reply:
added semver pattern validation. ref: https://semver.org/#is-there-a-suggested-regular-expression-regex-to-check-a-semver-string